packages/renderer/src/lib/ui/NumberInput.svelte

```diff
@@ -17,13 +17,14 @@ export let onValidation = (_value: number, _error?: string) => {};
 
 let minimumEnabled: boolean;
 let maximumEnabled: boolean;
+$: valueAsString = String(value);
 
-$: if (value !== undefined || disabled) {
+$: if (valueAsString !== undefined || disabled) {
```

Comment:
Maybe check if value is undefined when you convert it to String? I think that String(undefined) = "undefined"